modules/standard/IO.chpl

```diff
@@ -3832,11 +3832,13 @@ proc channel.readHelper(ref args ...?k, style:iostyleInternal):bool throws {
   :arg amount: The maximum amount of bytes to read.
   :returns: true if the bytes were read without error.
 */
+
+deprecated "channel.readline is deprecated. Use :proc:`channel.readLine` instead"
 proc channel.readline(arg: [] uint(8), out numRead : int, start = arg.domain.lowBound,
                       amount = arg.domain.highBound - start + 1) : bool throws
-                      where arg.rank == 1 && arg.isRectangular() && !arg.stridable {
+                      where arg.rank == 1 && arg.isRectangular() {
```

Comment:
I think this is supposed to still have `!arg.stridable`

Reply:
I think the reasoning to revert back to this was the “old are still broken, but no worse, and going away, so who cares; anyone switching to the new will be warned”. Tagging @bradcray to make sure I didn't misunderstand.

Reply:
OK that's fine if that's what Brad suggested. I was just checking it wasn't a merge error.

Reply:
I didn't see this before the merge, and hadn't thought of it staying or going, but I think it's fine either way.  If people were using ranges with non-unit stride with readline, it wasn't working right for them anyway (so they probably weren't).  And as soon as they get the deprecation warning, they'll hopefully switch over to readLine and be on a good path here on out.

This conversation makes me realize one thing which is that `stridable=true` doesn't necessarily equate to `stride != 1` which is how I treated it when adding the compiler errors for these overloads.  So I suppose there's a chance that some code would be using `readline()` with a `stridable=true, stride=1` array where their code would stop working due to the new error overloads.  In which case Shreyas backing this out is nice because it will give them something they can keep using for the time being if so (and hopefully they would complain to us).  But it does make me wonder whether we should switch from compilerError to throwing an illegalArgumentError or featureNotYetImplementedError or something like that instead.  Hmm...